arangod/Aql/OptimizerRules.cpp

```diff
@@ -7510,7 +7512,8 @@ void arangodb::aql::geoIndexRule(Optimizer* opt,
     }
 
     // if info is valid we try to optimize ENUMERATE_COLLECTION
-    if (info && info.collectionNodeToReplace == node) {
+    if (info && info.collectionNodeToReplace == node &&
+        !plan->hasForcedIndexHints()) {
```

Comment:
I think this is not right. This will disable the geo optimization if there are **any** indexHints in the query. I think we should check the indexHints for the current EnumerateCollectionNode only. Additionally, the indexHint for it may even refer to a geo index...